### ISSUE TITLE
Incorrect `unreachable_code` lint for loops in `try_blocks`

### ISSUE THREAD

**Issue by varkor** [rust-lang/rust#63324]
```rust
#![feature(try_blocks)]

pub fn foo() {
    let _: Result<(), ()> = try {
        loop {
            // anything here
        }
    };
}
```
Here, the entire `loop` expression is reported unreachable, despite not being so.
This is similar to https://github.com/rust-lang/rust/issues/54165, and probably has the same underlying issue.

**Comment by jonas-schievink** [comment]
Looks very similar to https://github.com/rust-lang/rust/issues/54165
